### PR TITLE
[Storage] Change truth value checking `ElementTree` to not emit `DeprecationWarning`

### PR DESCRIPTION
**Pull request by vincenttran-msft**
As title states! 

Before merging:

- Make change in `_shared` of other packages✔️

- Run live tests

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/response_handlers.py [hunk 1]
```diff
@@ -101,7 +101,8 @@ def process_storage_error(storage_error) -> NoReturn:  # pylint:disable=too-many
     try:
         error_body = ContentDecodePolicy.deserialize_from_http_generics(storage_error.response)
         try:
-            error_body = error_body or storage_error.response.reason
+            if error_body is None or len(error_body):
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/response_handlers.py:104]: Let's be explicit with the `len` check? `len(error_body) == 0`

**Review by jalauzon-msft — APPROVED** [review]

**Comment by akx** [comment]
Any idea when this might be released? Thanks! 😄

**Comment by jalauzon-msft** [comment]
Hi @akx, this will be included in our next release which is tentatively planned for beta in early February and a full release in March. Thanks.

**Pull request merged**